src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -679,6 +679,9 @@ public override Task WriteAsync(byte[] array, int offset, int count, Cancellatio
             }
         }
 
+        public override void CopyTo(Stream destination, int bufferSize) =>
+            CopyToAsync(destination, bufferSize).GetAwaiter().GetResult();
```

Comment:
This is more likely to be a step backward than forward: it's now likely going to block a whole thread while relying on the copy to be done on thread pool threads.